projects/plugins/jetpack/modules/cloudflare-analytics/class-wp-cloudflare-analytics.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Module Name: Cloudflare Analytics
+ * Module Description: Let WPCOM users automatically insert a Cloudflare analytics JS snippet into their site header.
+ * Requires Connection: Yes
+ * Auto Activate: No
+ *
+ * @package automattic/jetpack
+ */
+
+/**
+ * Bail if accessed directly
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WP_Cloudflare_Analytics class.
+ */
+class WP_Cloudflare_Analytics {
+
+	/**
+	 * Manages the insertion of Cloudflare analytics snippets into a user's
+	 * site <head>.
+	 *
+	 * @var WP_Cloudflare_Analytics Static property to hold our singleton instance
+	 */
+	private static $instance = false;
+
+	/**
+	 * Constructor method
+	 * Causes Cloudflare JS snippet to be included during head rendering.
+	 */
+	public function __construct() {
+		add_action( 'wp_head', array( $this, 'insert_code' ), 999 );
+	}
+
+	/**
+	 * Function to instantiate our class and make it a singleton
+	 */
+	public static function get_instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * This injects Cloudflare analytics code into the footer of the page.
+	 * Called exclusively by wp_head action
+	 */
+	public function insert_code() {
+		$option      = get_option( 'cloudflare_analytics' );
+		$tracking_id = isset( $option['code'] ) ? $option['code'] : '';
+		if ( empty( $tracking_id ) ) {
+			echo "<!-- Your Cloudflare Analytics Plugin is missing the tracking ID -->\r\n";
+			return;
+		}
+
+		// If we're in the admin_area, return without inserting code.
+		if ( is_admin() ) {
+			return;
+		}
+
+		$this->render_code( $tracking_id );
+	}
+
+	/**
+	 * Renders Cloudflare analytics code snippet.
+	 *
+	 * @param string $tracking_id Cloudflare Analytics tracking ID.
+	 */
+	private function render_code( $tracking_id ) {
+		printf(
+			"<!-- Cloudflare Web Analytics -->
+            <script defer
+                src='https://static.cloudflareinsights.com/beacon.min.js'
+                data-cf-beacon='{\"token\": \"%s\"}'>
+            </script>
+            <!-- End Cloudflare Web Analytics -->\r\n",
+			esc_html( $tracking_id )
+		);
+	}
+}
```

Comment:
I must say I'm not super sure I understand how you'd like to enable and use the feature on Atomic sites. I thought we could instantiate the class here, and conditionally require this file on all Atomic sites, either via a function in wpcomsh that would hook into the `jetpack_tools_to_include` filter to add the file, or by directly requiring the file in Jetpack's `projects/plugins/jetpack/modules/module-extras.php`. What do you think?

Reply:
We were going to directly require it in `wpcomsh`. It sounds like `jetpack_tools_to_include` would do the trick.

> or by directly requiring the file in Jetpack's projects/plugins/jetpack/modules/module-extras.php

We were holding off on doing this to avoid opening it up to standalone JP for now (of course, nothing stops someone from adding it themselves, which is fine). The partnership needs to get smoothed out before we open it up to all JP users (WP.com is basically a test bed at this point).

Reply:
I think that's alright. In my diff I basically limited things to Atomic sites with `jetpack_is_atomic_site()`, even if the file itself is required for everyone with a site connected to WordPress.com. We can remove that check once it's ready to ship to everyone.

Reply:
Just to make sure I'm understanding correctly - if we use the changes from your diff with `jetpack_is_atomic_site()` and `module-extras.php`, we will not need to make a change to wpcomsh?